src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -0,0 +1,1563 @@
+            final int bits = (0xff & src[i + srcPos]) << shift;
+            final int mask = 0xff << shift;
+            out = (out & ~mask) | bits;
+        }
+        return out;
+    }
+
+    /**
+     * <p>
+     * Converts an array of byte into a short using the default (little endian, Lsb0) byte and
+     * bit ordering.
+     * </p>
+     *
+     * @param src the byte array to convert
+     * @param srcPos the position in {@code src}, in byte unit, from where to start the
+     *            conversion
+     * @param dstInit initial value of the destination short
+     * @param dstPos the position of the lsb, in bits, in the result short
+     * @param nBytes the number of bytes to convert
+     * @return a short containing the selected bits
+     * @throws NullPointerException if {@code src} is {@code null}
+     * @throws IllegalArgumentException if {@code (nBytes-1)*8+dstPos >= 16}
+     * @throws ArrayIndexOutOfBoundsException if {@code srcPos + nBytes > src.length}
+     */
+    public static short byteArrayToShort(final byte[] src, final int srcPos, final short dstInit, final int dstPos,
+            final int nBytes) {
+        if (src.length == 0 && srcPos == 0 || 0 == nBytes) {
+            return dstInit;
+        }
+        if ((nBytes - 1) * 8 + dstPos >= 16) {
+            throw new IllegalArgumentException("(nBytes-1)*8+dstPos is greater or equal to than 16");
+        }
+        short out = dstInit;
+        for (int i = 0; i < nBytes; i++) {
+            final int shift = i * 8 + dstPos;
+            final int bits = (0xff & src[i + srcPos]) << shift;
+            final int mask = 0xff << shift;
+            out = (short) ((out & ~mask) | bits);
+        }
+        return out;
+    }
+
+    /**
+     * <p>
+     * Converts an array of Char into a long using the default (little endian, Lsb0) byte and
+     * bit ordering.
+     * </p>
+     *
+     * @param src the hex string to convert
+     * @param srcPos the position in {@code src}, in Char unit, from where to start the
+     *            conversion
+     * @param dstInit initial value of the destination long
+     * @param dstPos the position of the lsb, in bits, in the result long
+     * @param nHex the number of Chars to convert
+     * @return a long containing the selected bits
+     * @throws IllegalArgumentException if {@code (nHexs-1)*4+dstPos >= 64}
+     */
+    public static long hexToLong(final String src, final int srcPos, final long dstInit, final int dstPos,
+            final int nHex) {
+        if (0 == nHex) {
+            return dstInit;
+        }
+        if ((nHex - 1) * 4 + dstPos >= 64) {
+            throw new IllegalArgumentException("(nHexs-1)*4+dstPos is greater or equal to than 64");
+        }
+        long out = dstInit;
+        for (int i = 0; i < nHex; i++) {
+            final int shift = i * 4 + dstPos;
+            final long bits = (0xfL & hexDigitToInt(src.charAt(i + srcPos))) << shift;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).